### PR TITLE
test(use-visible-task): add more use visible task tests

### PR DESCRIPTION
**Pull request by Varixo**
# Overview

# What is it?

- [x] Feature / enhancement
- [ ] Bug
- [x] Docs / tests / types / typos

# Description

This PR adds more use visible task tests with fixes for v2

# Use cases and why

- 1. One use case
- 2. Another use case

# Checklist:

- [ ] My code follows the [developer guidelines of this project](https://github.com/BuilderIO/qwik/blob/main/CONTRIBUTING.md)
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] Added new tests to cover the fix / functionality

### CODE REVIEW

**Comment by gioboa** [comment]
@Varixo 👏 great

File: packages/qwik/src/core/v2/client/vnode-diff.ts [hunk 1]
```diff
@@ -702,7 +721,7 @@ export const vnode_applyJournal = (journal: VNodeJournalEntry[]) => {
         while (typeof (key = journal[idx] as string | null) === 'string') {
           idx++;
           const value = journal[idx++] as string | null;
-          if (key.startsWith(':')) {
+          if (key.startsWith(':') || key.startsWith('document:') || key.startsWith('window:')) {
```
Comment (mhevery) [inline packages/qwik/src/core/v2/client/vnode-diff.ts:724]: So this is not right. The code needs to stay as is, which means the code above needs to change.

The way it works now is that any property that starts with `:` is not serializable. 

For example `:onClick$` So i think we need to have `:document:onClick$` or something like that.

File: packages/qwik/src/core/v2/client/vnode-diff.ts [hunk 2]
```diff
@@ -567,9 +573,22 @@ export const vnode_diff = (container: ClientContainer, jsxNode: JSXOutput, vStar
       const element = vnode_getNode(vnode) as QElement2;
       if (!element.qDispatchEvent) {
         element.qDispatchEvent = (event: Event) => {
-          const eventName = event.type;
+          let eventName = event.type;
```
Comment (mhevery) [inline packages/qwik/src/core/v2/client/vnode-diff.ts:576]: I don't think this is right. See comment bellow about it having to start with `:`

**Review by mhevery — CHANGES_REQUESTED** [review]
Great work!  the only issue I seed is the `:` in the `vnode_getProp` function. 

Alos since you are adding `document` and `window` prefix, can we add `v2/use-on.unit.tsx` tests?


Otherwise this looks great!

**Pull request merged**